src/docs/arch-block.dox

```diff
@@ -1,9 +1,132 @@
 /*! @arch_page arch-block Block Manager
 
-The Block Manager manages the reading and writing of disk blocks
-in WiredTiger.  It does compression and encryption when these
-are configured.
+The WiredTiger block manager subsystem manages the reading and writing of data
+from the disk. It is designed to facilitate high performance, economic use of
+disk space and customisability.
+
+@section block What is a block?
+
+The block manager API introduces the concept of a "block" which represents a
+chunk of data that is stored on the disk and operated on as a single atomic
+unit. Each WiredTiger data file (any file in the home directory with the \c .wt
+suffix) is made up of these blocks.
+
+Since a data file can contain page images from multiple reconciliations of the
+same page, each block is associated with a single checkpoint. The API is
```

Comment:
A block can appear in any number of checkpoints, that is, writing a new checkpoint doesn't rewrite all of the blocks it contains.